Add inline code custom plugin

Bondok6 opened the pull request:
The issue: 

Users have to click on Insert HTML or switch to the Source Mode and create something like this:

```<p>Some text <code>inline code</code> more text</p>inline code```

In this PR:

I added a new custom plugin called `inline code`. This plugin provides a more user-friendly approach for adding inline code within the content. 
Users can now select the desired text and click on the `Insert Inline Code` icon to add inline code to their content


[Screencast from 13 يول, 2023 EEST 10:54:00 ص.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/116ee49b-ba5c-47a4-a39e-c007312c44ea)

Bondok6 commented:
Update: I changed the position of the `insert inline code` button to be next to the `insert code block` button

![image](https://github.com/Helpjuice/ckeditor5/assets/45231276/e0f54758-013c-4617-b6d4-ead411c2a4a4)

NicholasJarr commented on packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodecommand.ts:
`AttributeCommand` already exists in CKEditor. Is there any difference between this implementation and the CKEditor one?

If not, I think we can simplify this by removing this command and registering an instance of `AttributeCommand` instead:

```
editor.commands.add( INLINECODE, new AttributeCommand( editor, INLINECODE ) );
```

Bondok6 replied:
There's no difference.
I removed this file and imported `AttributeCommand` from the CKEditor

NicholasJarr commented on packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeediting.ts:
```diff
@@ -0,0 +1,30 @@
+			}
+		} );
+
+		editor.commands.add( INLINECODE, new InlineCodeCommand( editor, INLINECODE ) );
```
Minor, but I would use a different constant than the model name.

Bondok6 replied:
Done

Bondok6 commented:
> I tested it and it worked without issues 👍🏻 . Just left some suggestions.
> 
> BTW, is there a different icon for this? Maybe something with a code icon?

Thanks @NicholasJarr for your suggestions
 
- I changed the icon and added a shortcut "Ctrl + Alt + C"


[Screencast from 17 يول, 2023 EEST 09:19:22 ص.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/661cf1b4-4942-42f2-9793-503cfa5f0d97)

NicholasJarr commented on packages/ckeditor5-build-helpjuice-v2/src/plugins/inlinecode/inlinecodeui.ts:
```diff
@@ -0,0 +1,46 @@
+import { ButtonView } from 'ckeditor5/src/ui';
+import type AttributeCommand from '@ckeditor/ckeditor5-basic-styles/src/attributecommand';
+
+const INLINECODE = 'inlineCode';
+const INLINECODE_COMMAND = 'inlineCodeCommand';
```
Minor, but can we reuse the definitions from `inlinecodeediting.ts`. This way, we need to change only one place in order to update the constant value.

Bondok6 replied:
Done

NicholasJarr reviewed (commented):
@Bondok6

I tried it out and I noticed a small issue. If I add a inline code, change to the source editor, and then I switch back to the editor view, I can't toggle the inline code anymore:

https://github.com/Helpjuice/ckeditor5/assets/28270050/78e2f288-eca7-471f-a8f9-cdfbb6bfdd60

This is the code I'm using:

```html
<p>
    testing this stuff: <code>console.log('test');</code> .
</p>
```

Bondok6 commented:
> @Bondok6
> 
> I tried it out and I noticed a small issue. If I add a inline code, change to the source editor, and then I switch back to the editor view, I can't toggle the inline code anymore:
> 
>  2023-07-17.12-45-00.mp4 
> This is the code I'm using:
> 
> ```
> <p>
>     testing this stuff: <code>console.log('test');</code> .
> </p>
> ```

Hey @NicholasJarr , you have to **select the text** again and then click on the icon to toggle it 
That's how it works. ( in the same way for "Bold, Italic, ..." plugins )

> The text should be selected

[Screencast from 17 يول, 2023 EEST 11:32:00 م.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/027bc22f-5ca3-4048-b1e0-3539621f34cc)

NicholasJarr reviewed (commented):
> Hey @NicholasJarr , you have to **select the text** again and then click on the icon to toggle it That's how it works. ( in the same way for "Bold, Italic, ..." plugins )

My bad. I thought it worked differently.

I have just one last suggestion. We can't remove it through the Remove Format feature (like we can with bold, italics, etc.). Can you change Remove Format to allow it?

If I'm not mistaken, you only need to add this code after the call for `schema.extend` in `inlinecodeediting.ts`:

```js
 editor.model.schema.setAttributeProperties( INLINECODE, {
        isFormatting: true
    } );
```

Can you check if that works?

Bondok6 commented:
@NicholasJarr That's a good point! 
Let me try to add it.

Bondok6 commented:
@NicholasJarr It works well, thanks

[Screencast from 18 يول, 2023 EEST 05:34:08 م.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/65781d32-f4eb-4621-a428-9d4780166291)
